src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests2.cs

```diff
@@ -5545,5 +5545,1811 @@ static class E
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+
+        var e = comp.GetMember<NamedTypeSymbol>("E");
+        var extension = (SourceNamedTypeSymbol)e.GetTypeMembers().Single();
+        AssertEx.Equal("extension(System.ValueTuple`8<System.Int32, System.Int32, System.Int32, System.Int32, " +
+            "System.Int32, System.Int32, System.Int32, System.ValueTuple`1<System.String>>)",
+            extension.ComputeExtensionGroupingRawName());
+    }
+
+    [Fact]
+    public void GroupingTypeRawName_10()
+    {
+        // Simple types
+        var src = """
+static class E
+{
+    extension(C<char, string, bool, sbyte, short, int, long, float, double, byte, ushort, uint, ulong>)
+    {
+    }
+}
+class C<T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12> { }
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+
+        var e = comp.GetMember<NamedTypeSymbol>("E");
+        var extension = (SourceNamedTypeSymbol)e.GetTypeMembers().Single();
+        AssertEx.Equal("extension(C`13<System.Char, System.String, System.Boolean, System.SByte, " +
+            "System.Int16, System.Int32, System.Int64, System.Single, System.Double, " +
+            "System.Byte, System.UInt16, System.UInt32, System.UInt64>)",
+            extension.ComputeExtensionGroupingRawName());
+    }
+
+    [Fact]
+    public void GroupingTypeRawName_11()
+    {
+        // Native ints
+        var src = """
+static class E
+{
+    extension((nint, nuint))
+    {
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+
+        var e = comp.GetMember<NamedTypeSymbol>("E");
+        var extension = (SourceNamedTypeSymbol)e.GetTypeMembers().Single();
+        AssertEx.Equal("extension(System.ValueTuple`2<System.IntPtr, System.UIntPtr>)", extension.ComputeExtensionGroupingRawName());
+    }
+
+    [Fact]
+    public void GroupingTypeRawName_12()
+    {
+        // System.Nullable
+        var src = """
+static class E
+{
+    extension(int?)
+    {
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+
+        var e = comp.GetMember<NamedTypeSymbol>("E");
+        var extension = (SourceNamedTypeSymbol)e.GetTypeMembers().Single();
+        AssertEx.Equal("extension(System.Nullable`1<System.Int32>)", extension.ComputeExtensionGroupingRawName());
+    }
+
+    [Fact]
+    public void GroupingTypeRawName_13()
+    {
+        // Referencing type parameter
+        var src = """
+static class E
+{
+    extension<U>(U)
+    {
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+
+        var e = comp.GetMember<NamedTypeSymbol>("E");
+        var extension = (SourceNamedTypeSymbol)e.GetTypeMembers().Single();
+        AssertEx.Equal("extension<0>(!0)", extension.ComputeExtensionGroupingRawName());
+
+        // "0" instead of "T0"
+        var ilSrc = """
```

Comment:
Consider removing code from here to the end of the test. It looks like a WIP artifact.

Reply:
It looks like there are artifacts like that in other tests too. I do not think we should be testing ```CompileIL``` behavior. Even if it changes, I don't think we will need to do anything about it.